### PR TITLE
Fix public int read() throws IOException method exceeds the limit of length

### PR DESCRIPTION
**Pull request by xiaoheng1**
i use netty4.1.34

souce code:

```
public int read() throws IOException {
        if (!buffer.isReadable()) {
            return -1;
        }
        return buffer.readByte() & 0xff;
    }
```
In my opinion, the buffer.isReadable() method should not be used here to judge, but the available() method should be used. Because ByteBufInputStream is passed in length when constructing, so if the buffer.isReadable() method is used here If judged, it may exceed the limit of length, which is unreasonable.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@xiaoheng1 can you please add a test case ?

File: buffer/src/main/java/io/netty/buffer/ByteBufInputStream.java [hunk 1]
```diff
@@ -163,7 +163,8 @@ public boolean markSupported() {
 
     @Override
     public int read() throws IOException {
-        if (!buffer.isReadable()) {
+        int available = available();
+        if (available == 0) {
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/ByteBufInputStream.java]: nit: merge both lines

**Comment by xiaoheng1** [comment]
@normanmaurer I will add a test case test read method to the ByteBufStreamTest class.

File: buffer/src/test/java/io/netty/buffer/ByteBufStreamTest.java [hunk 2]
```diff
@@ -212,4 +212,34 @@ public void testReadLine() throws Exception {
...
+        buf.writeBytes(new byte[]{1, 2, 3, 4, 5, 6});
+
+        ByteBufInputStream in = new ByteBufInputStream(buf, 3);
+
+        assertEquals(1, in.read());
+        assertEquals(2, in.read());
+        assertEquals(3, in.read());
+        assertEquals(-1, in.read());
+        assertEquals(-1, in.read());
+        assertEquals(-1, in.read());
```
Comment (normanmaurer) [inline buffer/src/test/java/io/netty/buffer/ByteBufStreamTest.java]: Call buf.release() and in.close()

File: buffer/src/test/java/io/netty/buffer/ByteBufStreamTest.java [hunk 3]
```diff
@@ -212,4 +212,34 @@ public void testReadLine() throws Exception {
...
+
+        ByteBufInputStream in2 = new ByteBufInputStream(buf2, 4);
+
+        assertEquals(1, in2.read());
+        assertEquals(2, in2.read());
+        assertEquals(3, in2.read());
+        assertEquals(4, in2.read());
+        assertNotEquals(5, in2.read());
+        assertEquals(-1, in2.read());
+
```
Comment (normanmaurer) [inline buffer/src/test/java/io/netty/buffer/ByteBufStreamTest.java]: Call buf2.release() and in2.close()

Comment (xiaoheng1) [reply buffer/src/test/java/io/netty/buffer/ByteBufStreamTest.java]: @normanmaurer ok, i add it.

**Comment by normanmaurer** [comment]
@netty-bot Test this please

**Comment by normanmaurer** [comment]
@xiaoheng1 please fix the check style error:

```
07:31:16 [INFO] --- maven-checkstyle-plugin:3.0.0:check (check-style) @ netty-buffer ---
07:31:17 [INFO] Starting audit...
07:31:17 [ERROR] /code/buffer/src/test/java/io/netty/buffer/ByteBufStreamTest.java:248: an empty line before } [RegexpMultiline]
07:31:17 Audit done.
```

**Comment by normanmaurer** [comment]
@netty-bot Test this please

**Comment by normanmaurer** [comment]
@xiaoheng1 can you please sign our ICLA and let me know once done:

https://netty.io/s/icla

**Comment by xiaoheng1** [comment]
@normanmaurer I have already signed.

**Comment by normanmaurer** [comment]
@xiaoheng1 thanks a lot!

**Pull request merged**